app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,39 @@
+class SessionsController < Devise::SessionsController
+  layout 'minimal', :only => [ :timeout_check ]
+
+  def timeout_check
+    return if !session || !session["warden.user.user.session"]
```

Comment:
What value does warden.user.user.session store? Is it a timestamp? Also can we change the name of the key to something more clear?

Reply:
Sounds good; I added this:
```
# session["warden.user.user.session"] comes from adding timeoutable to the user model
# example: {"last_request_at"=>1537283353}
```